### ISSUE TITLE
Evaluate and update existing KPI metrics for visualization suitability

### ISSUE THREAD

**Issue by aakankshaduggal** [aicoe-aiops/ocp-ci-analysis#260]
As a Data Scientist working on the visualization of KPI metrics, I would want to :
* Collect more metrics in order to visualize them in a jupyter notebook environment. For example: for metrics like flake severity, the dataframe stored in ceph does not contain the metric and instead contains boolean values. In such case, it's better to store the metric correctly so we can access it in the visualization notebook.
* Go over all metrics notebook and see what all metrics need to be re-recorded.
* For a given, dashboard and job we would want to visualize all available metrics.

**Comment by MichaelClifford** [comment]
> In such case, it's better to store the metric correctly so we can access it in the visualization notebook.

@aakankshaduggal What is wrong with calculating this value (flake severity) in the visualization notebook? If we wait until the visualization notebook we can compute this value on the fly for different levels of analysis, like per build, per grid, per test in a grid, per test across the platform. If we store the computed values, we will have to have separate datasets for each of these levels of analysis. 

So, given the simpler and flexible data storage approach and relatively cheap computation (# of flakes / # of tests) its unclear to me why we can't generate this value in the visualization notebook.  Let me know if I'm missing something : ) 

>Go over all metrics notebook and see what all metrics need to be re-recorded.

Besides flake severity, what other values do you think need to be included in the ceph data set?

**Comment by aakankshaduggal** [comment]
Hey @MichaelClifford, for the flake severity notebook, the dataset we are storing in ceph contains only a boolean value. It might be interesting to store the flake severity value as well so as to visualize it too. 
Besides this, for the persistent_failure_analysis notebook, we have various metrics but do not have any timestamps to plot a time series for the same. 
For this issue, we will aim to find more such metrics that would be interesting to visualize.
cc : @chauhankaranraj @hemajv @isabelizimm

**Comment by hemajv** [comment]
@MichaelClifford We mainly wanted to use this issue to decide if we should revisit some of the KPI notebooks and figure out if we want to append additional metric values to the parquet files being stored in ceph. As Aakanksha pointed out, there are some KPI notebooks for which certain columns like `timestamp` are not being stored in the final parquet file and these columns are needed for the purpose of our visualizations, so we were hoping to track/note down such cases in this issue.

**Comment by MichaelClifford** [comment]
> It might be interesting to store the flake severity value as well so as to visualize it too

It's still unclear to me why we would want to store an additional dataset in ceph when "flake severity" can be calculated from the number_of_flakes boolean dataset using the two following two lines: 
```
flake_severity = df.groupby("test").flake.mean().reset_index()
flake_severity.rename(columns={"flake": "flake_severity"}, inplace=True)
```
> we have various metrics but do not have any timestamps to plot a time series for the same.

each parquet file name generated should include the `year-month-day` it was made and when the values were calculated. 

 `f"{METRIC_NAME}-{timestamp.year}-{timestamp.month}-{timestamp.day}.parquet",`

This could be used to determine the time period for specific data points. 

I'm not sure we'll want to add a "timestamp" column to each of these files though since it will be fairly repetitive (the same for every entry in a file) as these are summary statistics about the platform over time. In the example provided, persistent_failures_analysis, we are calculating rates and averages which occur over time, so what would the timestamps be here other then the change in these values from day to day (parquet to parquet)?

**Comment by aakankshaduggal** [comment]
> > It might be interesting to store the flake severity value as well so as to visualize it too
> 
> It's still unclear to me why we would want to store an additional dataset in ceph when "flake severity" can be calculated from the number_of_flakes boolean dataset using the two following two lines:
> 
> ```
> flake_severity = df.groupby("test").flake.mean().reset_index()
> flake_severity.rename(columns={"flake": "flake_severity"}, inplace=True)
> ```
Okay awesome, we could definitely do that :+1: 
> 
> > we have various metrics but do not have any timestamps to plot a time series for the same.
> 
> each parquet file name generated should include the `year-month-day` it was made and when the values were calculated.
> 
> `f"{METRIC_NAME}-{timestamp.year}-{timestamp.month}-{timestamp.day}.parquet",`
> 
> This could be used to determine the time period for specific data points.
> 
> I'm not sure we'll want to add a "timestamp" column to each of these files though since it will be fairly repetitive (the same for every entry in a file) as these are summary statistics about the platform over time. In the example provided, persistent_failures_analysis, we are calculating rates and averages which occur over time, so what would the timestamps be here other then the change in these values from day to day (parquet to parquet)?

here we are referring to a plot where we can have "change in the mean time to fix over time", so basically looking for the timestamp column that was a part of the data frame earlier. But if it doesn't look super important to us, we can definitely skip this :)

**Comment by isabelizimm** [comment]
Adding another unruly KPI :) `pct_fixed` seems to be NaN for all values, in all dashboards+jobs. It may be worthwhile to go back to the original notebook and evaluate if 1) tests don't get fixed or 2) this is being recorded correctly.

**Comment by MichaelClifford** [comment]
>We mainly wanted to use this issue to decide if we should revisit some of the KPI notebooks and figure out if we want to append additional metric values to the parquet files being stored in ceph. As Aakanksha pointed out, there are some KPI notebooks for which certain columns like timestamp are not being stored in the final parquet file and these columns are needed for the purpose of our visualizations, so we were hoping to track/note down such cases in this issue. 

I think this issue is a little too broad, and I would prefer if we could close it in favor of opening smaller issues for specific metric problems that can be tracked and fixed as we discover them. For example:  

> pct_fixed seems to be NaN for all values

we open an issue for it and it was addressed by #265 

If we don't have a solution for the "flake severity" metric, let's open a separate issue for that one too and any others that need to be updated.  

WDYT? @hemajv @aakankshaduggal

**Comment by aakankshaduggal** [comment]
Sure @MichaelClifford. We can close this issue and address the metrics separately :+1:
